### ISSUE TITLE
Add "Edit in GitHub" links at the end of each file

### ISSUE THREAD

**Issue by claremacrae** [obsidian-community/obsidian-hub#154]
The script should:

1. Loop through all markdown files in the hub, and see if they have the ` %% Hub footer: Please don't edit anything below this line %%` line
    1. If they do, replace anything below it with the contents of the jinja template
    2. If they don't append the ` %% Hub footer: Please don't edit anything below this line %%` line and the contents of the template

**Comment by argenos** [comment]
I finally had some time to dig out some of the work I had been doing on this. The template for the footer that should be added to all notes in the vault (after discussing a bit with @SlRvb what was the best way to target them with CSS to add icons) looked like this:

```jinja

 %% Hub footer: Please don't edit anything below this line %%
 
 
-[<span class="github-edit-note">Edit In Github</span>](https://github.dev/obsidian-community/obsidian-hub/blob/{{ file_path }})
+<span class="github-edit-note">[Edit In Github](https://github.dev/obsidian-community/obsidian-hub/blob/{{ file_path }})</span>
 
-<a class="github-copy-note" href="https://raw.githubusercontent.com/obsidian-community/obsidian-hub/main/{{ file_path }}" target="_blank" type="text/markdown" download>Copy this note</a>
+<span class="github-copy-note"> [Copy this note](https://raw.githubusercontent.com/obsidian-community/obsidian-hub/main/{{ file_path }})</span>
 
-<a class="github-vault-download" href="https://github.com/obsidian-community/obsidian-hub/archive/refs/heads/main.zip" download>Download this vault</a>
+<span class="github-vault-download">[Download this vault](https://github.com/obsidian-community/obsidian-hub/archive/refs/heads/main.zip)</span>


```

The example for the start page looked like this:

```jinja

 %% Hub footer: Please don't edit anything below this line %%
 
 
-[<span class="github-edit-note">Edit In Github</span>](https://github.dev/obsidian-community/obsidian-hub/blob/00%20-%20Start%20here.md)
+<span class="github-edit-note">[Edit In Github](https://github.dev/obsidian-community/obsidian-hub/blob/00%20-%20Start%20here.md)</span>
 
-<a class="github-copy-note" href="https://raw.githubusercontent.com/obsidian-community/obsidian-hub/main/00 - Start here.md" target="_blank" type="text/markdown" download>Copy this note</a>
+<span class="github-copy-note"> [Copy this note](https://raw.githubusercontent.com/obsidian-community/obsidian-hub/main/00%20-%20Start%20here.md)</span>
 
-<a class="github-vault-download" href="https://github.com/obsidian-community/obsidian-hub/archive/refs/heads/main.zip" download>Download this vault</a>
+<span class="github-vault-download">[Download this vault](https://github.com/obsidian-community/obsidian-hub/archive/refs/heads/main.zip)</span>


```

Note that some links needed to be escaped with `%20`, I can't remember during testing in which cases this was true or if it was still needed with the use of spans. 

Just to double-check, @SlRvb can you confirm this was the right/the latest version to target the links with CSS?

**Comment by SlRvb** [comment]
Yeah that was the right version to target them. I remember have some trouble making them pop _out_ in a way that would fit the screen nicely without looking weird on other devices, but that's more a css limitation so I'm still trying to figure out a good way to style it. 🤔

**Comment by dbarenholz** [comment]
I'm willing to implement this if someone can write 1 line on what the "jinja template" is, and where in the source I can find it.

**Comment by argenos** [comment]
A very summarized guide:
1. Create a file with the contents of my first comment in `.github/templates`
2. You'd need to get the template as it's done here (ignoring the file extension): https://github.com/obsidian-community/obsidian-hub/blob/d8b4e586fbce55226a4d474e5987c8a263322028/.github/scripts/update-releases.py#L54
3. You'd pass a dictionary with the variables for `{{ file_path }}`, e.g. 
    
    ```python
     footer = template.render(file_path=file_path_contents)
    ```

**Comment by claremacrae** [comment]
Closed via #234, #235, #236, #237, #238, #239, #240, #241 and #242
